### PR TITLE
Make Marathon's config file hierarchical

### PR DESCRIPTION
**Pull request by jsancio**

### CODE REVIEW

File: repo/packages/M/marathon/0/config.json [hunk 1]
```diff
@@ -1,147 +1,181 @@
...
+        "framework-name"
+      ],
+      "type": "object"
+    },
+    "mesos": {
+      "depscription": "Mesos specific configuration properties",
+      "properties": {
+        "master": {
+          "default": "zk://master.mesos:2181/mesos",
+          "description": "The URL of the Mesos master. The format is a comma-delimited list of hosts like zk://host1:port,host2:port/mesos. If using ZooKeeper, pay particular attention to the leading zk:// and trailing /mesos! If not using ZooKeeper, standard URLs like http://localhost are also acceptable.",
```
Comment (ConnorDoyle) [inline repo/packages/M/marathon/0/config.json]: Drop `http://` in the "no ZK" example.  If not using ZK, then the value can look like this:

- `localhost:5050`
- `10.0.0.5:5050,10.0.0.6:5050`

File: repo/packages/M/marathon/0/marathon.json [hunk 2]
```diff
@@ -1,21 +1,15 @@
...
+  "instances": {{marathon.instances}},
   "constraints": [["hostname", "UNIQUE"]],
   "ports": [
-    {{marathon/http-port}}
-
-    {{#marathon/https-port}}
-    , {{marathon/https-port}}
-    {{/marathon/https-port}}
+    {{marathon.http-port}}
+    , {{marathon.https-port}}
```
Comment (ConnorDoyle) [inline repo/packages/M/marathon/0/marathon.json]: If we swap the order of the libprocess port (always required) and the https port (only required if that protocol is enabled) then we can optionally avoid consuming a port we don't intend to bind here.

**Comment by ConnorDoyle** [comment]
Let's test-launch this on DCOS to make sure it comes up with the default parameters.

File: repo/packages/M/marathon/0/config.json [hunk 3]
```diff
@@ -1,147 +1,181 @@
...
+        },
+        "zk-timeout": {
+          "description": "Timeout for ZooKeeper in milliseconds.",
+          "minimum": 0,
+          "type": "integer"
+        }
+      },
+      "required": [
+        "zk",
+        "framework-name"
```
Comment (ConnorDoyle) [inline repo/packages/M/marathon/0/config.json]: These two properties already have defaults.  We should be consistent with how we mark properties required.  Two options:

1. Remove these properties from the required list to match the others.
2. Add the properties that we expand without conditional guards in `marathon.json` to the required list (my preference).

File: repo/packages/M/marathon/0/config.json [hunk 4]
```diff
@@ -1,147 +1,181 @@
...
+          "description": "Password for the keystore supplied with the marathon/ssl-keystore-path option.",
+          "type": "string"
+        },
+        "ssl-keystore-path": {
+          "description": "Path to the SSL keystore. SSL will be enabled if this option is supplied. The keystore can be downloaded into Marathon's working directory by using the uris property.",
+          "type": "string"
+        },
+        "uris": {
+          "default": [],
+          "description": "List of URIs that will be download and made available in the current working directory of Marathon. For example this can be used to download Java keystore file for SSL configuration.",
```
Comment (ConnorDoyle) [inline repo/packages/M/marathon/0/config.json]: Missing article: "..._a_ Java keystore file..."

File: repo/packages/M/marathon/0/config.json (hunk 3 above)
Comment (jsancio) [inline repo/packages/M/marathon/0/config.json]: I'll do 2).

File: repo/packages/M/marathon/0/marathon.json (hunk 2 above)
Comment (jsancio) [inline repo/packages/M/marathon/0/marathon.json]: Done.

File: repo/packages/M/marathon/0/config.json (hunk 3 above)
Comment (jsancio) [inline repo/packages/M/marathon/0/config.json]: Done.

File: repo/packages/M/marathon/0/config.json (hunk 4 above)
Comment (jsancio) [inline repo/packages/M/marathon/0/config.json]: Done.

**Comment by jsancio** [comment]
Tested with DCOS!

**Pull request merged**